robot-core/src/main/java/org/obolibrary/robot/ReasonOperation.java

```diff
@@ -112,6 +112,83 @@ public static void reason(
+    if (!structural) {
+      OWLOntology empty = OWLManager.createOWLOntologyManager().createOntology();
+      return new ReasonerFactory().createReasoner(empty);
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Given an OWLAxiom, a tautology checker reasoner, and a boolean, determine if the axiom is a
+   * tautology.
+   *
+   * @param axiom OWLAxiom to check
+   * @param tautologyChecker OWLReasoner for empty ontology
+   * @param structural if true, only check for hard-coded structural patterns (checker can be null)
+   * @return true if axiom is tautological
+   */
+  public static boolean isTautological(
+      OWLAxiom axiom, OWLReasoner tautologyChecker, boolean structural) {
+    if (structural) {
+      if (axiom instanceof OWLSubClassOfAxiom) {
+        OWLSubClassOfAxiom subClassOfAxiom = (OWLSubClassOfAxiom) axiom;
+        if (subClassOfAxiom.getSuperClass().isOWLThing()) {
+          // X subClassOf owl:Thing
+          return true;
+        } else if (subClassOfAxiom.getSubClass().isOWLNothing()) {
+          // owl:Nothing subClassOf X
+          return true;
+        } else {
+          // X subClassOf X
+          return subClassOfAxiom.getSubClass().equals(subClassOfAxiom.getSuperClass());
+        }
+      } else if (axiom instanceof OWLEquivalentClassesAxiom) {
+        // X equivalentTo X
+        OWLEquivalentClassesAxiom equivAxiom = (OWLEquivalentClassesAxiom) axiom;
+        return equivAxiom.getClassExpressions().size() < 2;
+      } else if (axiom instanceof OWLClassAssertionAxiom) {
+        // X a owl:Thing
+        OWLClassAssertionAxiom classAssertion = (OWLClassAssertionAxiom) axiom;
+        return classAssertion.getClassExpression().isOWLThing();
+      } else if (axiom instanceof OWLObjectPropertyAssertionAxiom) {
+        // X owl:topDataProperty ...
+        OWLObjectPropertyAssertionAxiom assertion = (OWLObjectPropertyAssertionAxiom) axiom;
+        return assertion.getProperty().isOWLTopObjectProperty();
+      } else if (axiom instanceof OWLDataPropertyAssertionAxiom) {
+        // X owl:topObjectProperty ...
+        OWLDataPropertyAssertionAxiom assertion = (OWLDataPropertyAssertionAxiom) axiom;
+        return assertion.getProperty().isOWLTopDataProperty();
+      } else if (axiom instanceof OWLDeclarationAxiom) {
+        // owl:Thing a owl:Class, owl:Nothing a owl:Class, etc.
+        OWLDeclarationAxiom declaration = (OWLDeclarationAxiom) axiom;
+        return declaration.getEntity().isBuiltIn();
+      }
+    } else if (tautologyChecker != null) {
+      if (axiom instanceof OWLAnnotationAssertionAxiom) {
+        // ignore annotation assertions
+        return false;
+      } else if (axiom instanceof OWLDeclarationAxiom) {
+        // ignore declaration UNLESS it is built in
+        OWLDeclarationAxiom declaration = (OWLDeclarationAxiom) axiom;
+        return declaration.getEntity().isBuiltIn();
+      }
+      // otherwise check if axiom is entailed by empty ontology
```

Comment:
Not sure it is necessary, but I would suggest to add a check here:

```
else if (axiom instanceof OWLLogicalAxiom) { ...}
```

I am not sure otherwise what would happen if for example irrelevant axioms (rules etc) are send to isEntailed(). Should not be a problem, but no harm here if you add it.